alchemy/src/docker/api/api.ts

```diff
@@ -0,0 +1,218 @@
+  M extends HttpMethod,
+  // @ts-ignore: dyncamically inferred on usage
+  K extends keyof NonNullable<paths[P][M]["parameters"]["path"]>,
+  // @ts-ignore: dyncamically inferred on usage
+> = Pick<NonNullable<paths[P][M]["parameters"]["path"]>, K>;
+
+type PickBodyParam<
+  P extends keyof paths,
+  M extends HttpMethod,
+  K extends keyof NonNullable<
+    // @ts-ignore: dyncamically inferred on usage
+    paths[P][M]["requestBody"]["content"]["application/json"]
+  >,
+> = Pick<
+  // @ts-ignore: dyncamically inferred on usage
+  NonNullable<paths[P][M]["requestBody"]["content"]["application/json"]>,
+  K
+>;
+
+export function createDockerApi() {
+  const client = createClient<paths>({
+    baseUrl: "http://localhost:10000",
+    fetch(input) {
+      if (typeof Bun !== "undefined") {
+        // @ts-ignore: Bun specific
+        input.unix = "/var/run/docker.sock";
+      }
+
+      return fetch(input.url, input);
+    },
+  });
+
+  const GET = async <
+    T extends ClientPathsWithMethod<typeof client, "get">,
+    Init extends MaybeOptionalInit<paths[T], "get">,
+  >(
+    path: T,
+    ...init: InitParam<Init>
+  ): Promise<MethodResponse<typeof client, "get", T>> => {
+    // @ts-ignore: dyncamically inferred on usage
+    const { data, error, response } = await client.GET(path, ...init);
+
+    if (error) {
+      throw new Error(
+        `Failed to GET ${path}: ${response?.status} ${response?.statusText}: ${JSON.stringify(error, null, 2)}`,
+      );
+    }
+
+    return data as NonNullable<MethodResponse<typeof client, "get", T>>;
+  };
+
+  const POST = async <
+    T extends ClientPathsWithMethod<typeof client, "post">,
+    Init extends MaybeOptionalInit<paths[T], "post">,
+  >(
+    path: T,
+    ...init: InitParam<Init>
+  ): Promise<MethodResponse<typeof client, "post", T>> => {
+    // @ts-ignore: dyncamically inferred on usage
+
+    const { data, error, response } = await client.POST(path, ...init);
+
+    if (error) {
+      throw new Error(
+        `Failed to POST ${path}: ${response?.status} ${response?.statusText}: ${JSON.stringify(error, null, 2)}`,
+      );
+    }
+
+    return data as NonNullable<MethodResponse<typeof client, "post", T>>;
+  };
+
+  const DELETE = async <
+    T extends ClientPathsWithMethod<typeof client, "delete">,
+    Init extends MaybeOptionalInit<paths[T], "delete">,
+  >(
+    path: T,
+    ...init: InitParam<Init>
+  ): Promise<MethodResponse<typeof client, "delete", T>> => {
+    // @ts-ignore: dyncamically inferred on usage
+    const { data, error, response } = await client.DELETE(path, ...init);
+
+    if (error) {
+      throw new Error(
+        `Failed to DELETE ${path}: ${response?.status} ${response?.statusText}: ${JSON.stringify(error, null, 2)}`,
+      );
+    }
+
+    return data as NonNullable<MethodResponse<typeof client, "delete", T>>;
+  };
+
+  return {
+    containers: {},
+    network: (nameOrId: PickPathParam<"/networks/{id}", "get", "id">) => ({
```

Comment:
Better to return methods outside of the resource lifecycle handler because functions cannot be serialized.

When a resource is skipped, we don't invoke your function and instead return the persisted state data.

A wrapper function can then implement methods like `network` that use the data